pkg/upgraders/osd/upgrader.go

```diff
@@ -589,19 +589,34 @@ func performClusterHealthCheck(c client.Client, metricsClient metrics.Metrics, c
 		return false, fmt.Errorf("unable to query critical alerts: %s", err)
 	}
 
-	if len(alerts.Data.Result) > 0 {
-		logger.Info("There are critical alerts exists, cannot upgrade now")
-		return false, fmt.Errorf("there are %d critical alerts", len(alerts.Data.Result))
+	alertCount := len(alerts.Data.Result)
+
+	if alertCount > 0 {
+		alert := []string{}
+		uniqueAlerts := make(map[string]bool)
+
+		for _, r := range alerts.Data.Result {
```

Comment:
```suggestion
		for _, r := range alertCount {
```

Reply:
Well, `alerts.Data.Result` would be a slice of the type `metrics.AlertResult` which I need to iterate over to fetch each alert. `alertCount` is an integer for the total count so can't apply `range` over it. So it seems fine to me. Or did you mean anything else?

Reply:
ah sorry my bad here.